### DISCUSSION TITLE
Install on a light VPS but use external samba volume as storage

### DISCUSSION THREAD

**Discussion by iambenmitchell** [nextcloud/all-in-one#916] [Questions] [answered]
Hello,

I would like to install next cloud using docker on a small VPS (40gb) and then use a mounted samba/cifs server as storage (1TB).

I have the CIFS mounted in fstab as so:

```
//address_of_samba/backup /mnt/sam cifs credentials=/root/.smbcredentials 0 0
```

I have made a directory on the share for next cloud located at `/mnt/sam/nextcloud`

I have then set the permissions like so:

```
sudo chmod -R 750 /mnt/sam/nextcloud
sudo chown -R 33:0 /mnt/sam/nextcloud
```

The command I use to install nextcloud is this:

```
sudo docker run -it \
--name nextcloud-aio-mastercontainer \
--restart always \
-p 80:80 \
-p 8080:8080 \
-p 8443:8443 \
-e NEXTCLOUD_DATADIR="/mnt/sam/nextcloud" \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
```

After going through the setup process, it is time to start the containers. However nextcloud gets stuck on starting. The logs read this:

```
nc: getaddrinfo: Name does not resolve
Waiting for database to start...
Configuring Redis as session handler
The www-data user doesn't seem to have access rights in /mnt/ncdata.
Did you maybe change the datadir and did forget to apply the correct permissions?
touch: /mnt/ncdata/this-is-a-test-file: Permission denied
Configuring Redis as session handler
```

What am I doing wrong?

**Comment by szaimen** [comment, marked as answer]
> What am I doing wrong?

You are not applying the permissions correctly.

CIFS mounts are not able to handle chmod and chown. You need to set the correct permissions at mounting time. 

See e.g. https://github.com/nextcloud/vm/blob/961c3862dfffbf3dfc001b73bda90414e09123fb/apps/smbmount.sh#L663 for a config that should work.
